devtools/x/src/installer.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{cargo::Cargo, config::CargoConfig};
+use log::info;
+use std::process::Command;
+
+pub fn install_if_needed_or_die(name: &str, version: &str) {
```

Comment:
Is this only supposed to install a different cargo version? (I'm assuming yes)

Reply:
Don't want to use the nightly version that could cause breakage in these released version of tools.

Reply:
if we don't have the current cargo stable locally, totally fine with using that to build the latest tooling, since the stable version we're using isn't in x.toml -- though weirdly the nightly version is.   This all seems to need a reworking but not one I wanted to take on now with the redundant info in x.toml and cargo-toolchain.